File: client/app/src/main/java/com/rh/heji/data/BillType.java
package com.rh.heji.data;

/**
 * Date: 2020/11/20
 * Author: 锅得铁
 * #
 */
public enum BillType {

    INCOME(+1, "收入"),
    EXPENDITURE(-1, "支出");
    private int type;
    private String text;

    BillType(int type, String text) {
        this.type = type;
        this.text = text;
    }

    public int type() {
        return type;
    }

    public String text() {
        return text;
    }

    public static BillType transform(int type) {
        if (type == BillType.INCOME.type()) {
            return BillType.INCOME;
        }else {
            return BillType.EXPENDITURE;
        }
//        else if (type == BillType.EXPENDITURE.type()) {
//            return BillType.EXPENDITURE;
//        }
//        return null;
    }

    public static BillType transform(String text) {
        if (text.equals(BillType.INCOME.text())) {
            return BillType.INCOME;
        } else if (text.equals(BillType.EXPENDITURE.text())) {
            return BillType.EXPENDITURE;
        }
        return null;
    }
}

File: client/app/src/main/java/com/rh/heji/data/BillType.kt
package com.rh.heji.data

/**
 * @date: 2020/11/20
 * @author: 锅得铁
 * #
 */
enum class BillType(private val type: Int, private val text: String) {
    INCOME(+1, "收入"), EXPENDITURE(-1, "支出"), ALL(0, "收支");

    fun valueInt(): Int = type
    fun valueString(): String = text

    companion object {
        fun transform(type: Int): BillType {
            if (type == INCOME.valueInt()) {
                return INCOME
            }
            return if (type == EXPENDITURE.valueInt()) {
                EXPENDITURE
            } else {
                ALL
            }
        }

        fun transform(text: String): BillType {
            if (text == INCOME.valueString()) {
                return INCOME
            } else if (text == EXPENDITURE.valueString()) {
                return EXPENDITURE
            }
            return ALL
        }
    }

    override fun toString(): String {
        return "BillType(name=$name,type=$type,text=$text,ordinal=$ordinal)"
    }
}